standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/ObjectStore.java

```diff
@@ -8556,6 +8556,9 @@ public UpdateMDatabaseURIRetVal updateMDatabaseURI(URI oldLoc, URI newLoc, boole
             updateLocations.put(locationURI.toString(), dbLoc);
             if (!dryRun) {
               mDB.setLocationUri(dbLoc);
+              if (org.apache.commons.lang3.StringUtils.isNotBlank(mDB.getManagedLocationUri())) {
+                mDB.setManagedLocationUri(dbLoc);
```

Comment:
Looks like we are setting the value of locationURI for managedLocation. dbLoc is based on mDB.getLocationURI(). Instead we should have similar logic where string substitutions are performed on DB's managedlocation as well.